src/pages/CreateAssignment/CreateAssignment.js

```diff
@@ -49,9 +49,46 @@ function CreateAssignment(props) {
     setLessonCreationDate(originalDate);
   }, [firebase]);
 
+  async function parseCardsFromLesson(lesson) {
+    let lessonCards = {};
+    let deploymentAccountIds = new Set();
+    for (const dueDate in lesson.dueDates) {
+      for (const deploymentAccount of lesson.dueDates[dueDate]) {
+        deploymentAccountIds.add(deploymentAccount);
+      }
+    }
+
+    let deploymentNameMap = {};
+    Array.from(deploymentAccountIds).forEach(id => {
+      firebase
+        .getDeploymentAccountInformation(id)
+        .then(deploymentAccount => {
+          deploymentNameMap[id] = deploymentAccount.username;
+        })
+        .catch(error => console.error("Error getting custom lesson: ", error));
+    });
+
+    // Wait for deploymentNameMap to finish resolving
+    while (Object.keys(deploymentNameMap).length == 0) {
+      await new Promise(r => setTimeout(r, 500));
+    }
+
+    for (const dueDate in lesson.dueDates) {
+      let lessonCard = [[], []];
+
+      for (const deploymentAccountId of lesson.dueDates[dueDate]) {
+        lessonCard[0].push(deploymentAccountId); // inputting account id
+        lessonCard[1].push(deploymentNameMap[deploymentAccountId]); // input corresponding username
+      }
```

Comment:
i'm not a fan of using array indices instead of objects (i.e. having each lessonCard be `[{id: _d, username: username}])` - is there a reason it's written this way?

Reply:
this is to conform with prashant's implementation of dueDates in lessonCards